ahorro_combustible/utils.py

```diff
@@ -0,0 +1,57 @@
+from typing import Dict, Tuple, Sequence
+
+
+def count_brands(gas_stations) -> Sequence[Tuple[int, str]]:
+    '''Count all the brands in the `gas_stations` input. It can be considered that
+    the ``Rotulo`` entry contains the brand name of the gas station.
+
+    Parameters
+    ----------
+    gas_stations : dict
+        A dictionary that contains at least the equivalent structure to the
+        following example.
+
+        { 'ListaEESSPrecio': [ { 'Rótulo': 'ES DULANTZI REPSOL' }, ... ]}
+
+    Returns
+    -------
+    Sequence[Tuple[int, str]]
+        A sorted sequence of tuples being the first part of each tuple the count
+        number of the second part, the brand name.
+
+        The items will be sorted by the count number (highest, first) and the
+        brand name.
+
+    Examples
+    --------
+
+    >>> count_brands({'ListaEESSPrecio': [ {'Rótulo': 'A' },
+    ...                                    {'Rótulo': 'B' },
+    ...                                    {'Rótulo': 'A' }]})
+    [(2, 'A'), (1, 'B')]
+
+    >>> count_brands({'ListaEESSPrecio': [ {'Rótulo': 'A' },
+    ...                                    {'Rótulo': 'C' },
+    ...                                    {'Rótulo': 'C' },
+    ...                                    {'Rótulo': 'B' }]})
+    [(2, 'C'), (1, 'B'), (1, 'A')]
+    '''
+        
+    # Through this loop I take the brands and petrol station quantity
+    
+    pet_sts = [pet_st["Rótulo"] for pet_st in gas_stations["ListaEESSPrecio"]]        
+            
+    # Another empty list to add the final result without selection and order
+          
+    lst_tup_sort = []
+    
+    # Next loop: to number, remove duplicates and sort
+    
+    for pet_st in pet_sts:
+        count_pet_st = pet_sts.count(pet_st)
```

Comment:
To get the count of the `pet_st` variable the `List.count()` method needs to visit all the items of the `pet_sts` list (with size `n`). That this, the time complexity of this function is `O(n)`. Because this operation is inside the loop with complexity `O(n)` as well. That gives us an accumulate complexity of `O(n^2)`.